### PR TITLE
Delete session on submit

### PR DESCRIPTION
**Pull request by H3ll3m4**
[Trello](https://trello.com/c/QOUPgM0p/3281-delete-session-on-submit)

## Goal
Problem was being able to get the session after submitting the form. Even if the submission can be differentiated by reference number, it was still possible to re upload and change submission, this was a problem for the cop form where documents were uploaded by solicitors.

## Implementation
We would like to delete session after submission of the form data to the submitter. We use flash part of the session that is being purged after next request. We're using 2 steps so it doesn't redirect straight away to expired session page after submission. Limitation at the moment is the back button behaviour. It is still possible to get the session held in the client using the back button.

## Testing
Unit and manual testing locally and in test-dev.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -122,4 +126,8 @@ def payment_link_url
     ENV['PAYMENT_LINK'] + show_reference_number
   end
   helper_method :payment_link_url
+
+  def delete_session
+    flash[:confirmation] = 'Session will expired'
```
Comment (StevenLeighton21) [inline app/controllers/application_controller.rb:131]: Do we have a design for this? I don't see anything in the trello ticket, might be worth confirming with design/content what to display here

File: app/filters/verify_session.rb [hunk 2]
```diff
@@ -1,5 +1,12 @@
 class VerifySession
   def self.before(controller)
+    if !allowed_pages?(controller) && controller.flash[:expired_session].present?
+      controller.reset_session
+      controller.redirect_to '/session/expired'
+    end
+    if !allowed_pages?(controller) && controller.flash[:confirmation].present?
+      controller.flash[:expired_session] = 'Session has expired'
```
Comment (StevenLeighton21) [inline app/filters/verify_session.rb:8]: Same comment re: content might have ideas on this text. Also it should probably be retrieved from the en.yml config through the i18n module so we can internationalise this easily if we need to.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (H3ll3m4) [reply app/controllers/application_controller.rb:132]: As discussed, values not used to display as toggle that that disappeared by next request

File: app/filters/verify_session.rb (hunk 2 above)
Comment (H3ll3m4) [reply app/filters/verify_session.rb:8]: idem

**Review by StevenLeighton21 — APPROVED** [review]

**Pull request merged**